src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,19 +39,27 @@
           type="text"
           className="input"
           placeholder="Search..."
+          value={query}
+          onChange={event =>
+            dispatch(
+              filterSlice.actions.setQuery(event.target.value.trimStart()),
```

Comment:
Consider debouncing the input change handler to improve performance and reduce the number of dispatches when the user types quickly. This can be achieved using a custom hook or a utility function.